cmd/root.go

```diff
@@ -0,0 +1,111 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+
+	utils "github.com/fabric8-analytics/cli-tools/utils"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var cfgFile string
+var debug bool
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "crda",
+	Short: "Cli tool to interact with CRDA Platform.",
+	Long:  `Cli tool to interact with CRDA Platform. Perfoms synk token validation.`,
+	Args:  cobra.ExactValidArgs(1),
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+// Flag Defaults
+const (
+	Host      = "https://f8a-analytics-preview-2445582058137.production.gw.apicast.io:443"
+	SnykToken = ""
+	AuthToken = "3e42fa66f65124e6b1266a23431e3d08"
+	Debug     = false
+)
+
+func init() {
+	cobra.OnInitialize(initConfig)
+
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.crda/config.yaml)")
+	rootCmd.PersistentFlags().BoolVarP(&debug, "debug", "d", Debug, "activate panic")
+	rootCmd.PersistentFlags().String("host", Host, "Host Server, if set, host from config file will be ignored.")
+	rootCmd.PersistentFlags().String("snyk-token", SnykToken, "Snyk token, if not set, Freemium account will be created.")
+	rootCmd.PersistentFlags().String("auth-token", AuthToken, "3 Scale Token, Token for server authentication.")
+	viper.BindPFlag("debug", rootCmd.PersistentFlags().Lookup("debug"))
```

Comment:
debug shouldn't be persisted in a config, it is just a transient state which is applicable just for current command session. Basically it should just alter log level.

Reply:
Ack.